modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/OperationParameterValueResolver.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.runtime.operation;
+
+import org.mule.runtime.api.meta.model.operation.OperationModel;
+import org.mule.runtime.extension.api.runtime.operation.OperationContext;
+import org.mule.runtime.module.extension.internal.introspection.ParameterGroup;
+import org.mule.runtime.module.extension.internal.model.property.ParameterGroupModelProperty;
+import org.mule.runtime.module.extension.internal.runtime.ParameterValueResolver;
+import org.mule.runtime.module.extension.internal.runtime.resolver.ParameterGroupArgumentResolver;
+
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * {@link ParameterValueResolver} implementation for {@link OperationMessageProcessor}
+ *
+ * @since 4.0
+ */
+public class OperationParameterValueResolver implements ParameterValueResolver {
```

Comment:
1. Yes, It's a bad and old method name, should be removed.
2. Yes, uses the OperationContext to reuse logic as much as possible.
3. In the SDK doesn't exist the logic of resolve a `ParameterGroup` from the name of the container of it, so this logic is necessary.